amcl/src/amcl/sensors/amcl_laser.cpp

```diff
@@ -155,12 +155,21 @@ double AMCLLaser::BeamModel(AMCLLaserData *data, pf_sample_set_t* set)
 
   total_weight = 0.0;
 
+  int invalidSamples = 0;
+
   // Compute the sample weights
   for (j = 0; j < set->sample_count; j++)
   {
     sample = set->samples + j;
     pose = sample->pose;
```

Comment:
You could move this down after the isValidSample check since it isn't needed if you have an invalid sample.